packages/@aws-cdk/aws-appmesh/test/integ.mesh.ts

```diff
@@ -29,7 +34,7 @@ const virtualService = new appmesh.VirtualService(stack, 'service', {
 });
 
 const node = mesh.addVirtualNode('node', {
-  serviceDiscovery: appmesh.ServiceDiscovery.dns(`node1.${namespace.namespaceName}`),
+  serviceDiscovery: appmesh.ServiceDiscovery.dns(`node1.${namespace.namespaceName}`, undefined, appmesh.IpPreference.IPV4_ONLY),
```

Comment:
how does the IP preference of the `serviceDiscovery` in this node interact with the ip preference of the `serviceDiscovery` set on lines 21-23? Does one overwrite the other? Is this allowed? This behavior should be documented in the comment block of the method.

Reply:
> Please update this PR's description to describe the feature you're adding and give a high-level overview of how it works.

I've gone ahead and added some context. If further elaboration would help, feel free to let me know.

Reply:
There are two resource levels we allowing setting this preference as you point out. A Mesh is the top level resource which users will be creating Virtual Nodes within. A Mesh preference will apply to all Virtual Nodes as a default value and each individual Virtual Node can set its own preference to override the Mesh preference. If there is no preference set anywhere then no default preference is applied.